### PR TITLE
Fix some warnings

### PR DESCRIPTION
**Pull request by triller-telekom**
Also configure HTML validation to ignore unknown elements and attributes
which exists due to the fact that Eclipse doesn't know the ng-* etc. tags.

Relates to #5853

Signed-off-by: Stefan Triller <stefan.triller@telekom.de>

### CODE REVIEW

File: bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/ConfigUtilTest.java [hunk 1]
```diff
@@ -38,6 +38,8 @@
     }
 
     private static class L<T> extends ArrayList<T> {
```
Comment (htreu) [inline bundles/config/org.eclipse.smarthome.config.core.test/src/test/java/org/eclipse/smarthome/config/core/ConfigUtilTest.java]: please remove this inner class completely, use `Collections.singletonList(configDescriptionInteger)` or/and `Arrays.asList(..., ...)` instead.

Also: remove the method 'm' and replace by `Map<String, Object> fooMap = Collections.singletonMap("foo", "1");`.

File: bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/DimmerItemTest.java [hunk 2]
```diff
@@ -95,7 +95,7 @@ public void testAcceptedStates() {
     }
 
     private class DummyType extends HSBType {
-
+        private static final long serialVersionUID = -3327574976250687313L;
```
Comment (htreu) [inline bundles/core/org.eclipse.smarthome.core.test/src/test/java/org/eclipse/smarthome/core/library/items/DimmerItemTest.java]: what is the point of this class anyway? Testing the inheritance of of Java? Please remove.

**Comment by triller-telekom** [comment]
Thanks for the review. I have addressed all your comments and fixed the build problem. Travis should succeed now.

**Comment by triller-telekom** [comment]
FYI: Down from >5200 to 1435 warnings and ~15 info messages to 0.

**Comment by maggu2810** [comment]
Great! Thanks

**Pull request merged**